bigtable/integration_test.go

```diff
@@ -1243,6 +1248,88 @@ func TestIntegration_TableIam(t *testing.T) {
 	}
 }
 
+func TestIntegration_BackupIAM(t *testing.T) {
```

Comment:
Given this is an integration test file, it feels like we shouldn't be mocking out the interactions. It's probably enough to follow the pattern of `TestIntegration_TableIam` (which should be `IAM`, but that's another story :upside_down_face:).

Reply:
+1. @kolea2 do you know how other languages are handling this testing, and if we will face any quota-related issues?

Reply:
I added a mock setup because I got an error while comparing policies. I had an `Etag` mismatch, even in the read-modify-write cycle.
From [documentation](https://cloud.google.com/bigtable/docs/reference/admin/rest/v2/Policy): 
> It is strongly suggested that systems make use of the etag in the read-modify-write cycle to perform policy updates in order to avoid race conditions: An etag is returned in the response to getIamPolicy, and systems are expected to put that etag in the request to setIamPolicy to ensure that their change will be applied to the same version of the policy.

But I had and still have this error without mocking.
![image](https://user-images.githubusercontent.com/38891896/99732250-396de680-2ad0-11eb-9832-41cb3ca37acd.png)

Will wait for response!

P.S. Anyway, I can also rename `TestIntegration_TableIam` to `TestIntegration_TableIAM` here :)
It's related to this PR.

Reply:
The Etag should only change when there is a modification. So, something else must have modified the value between the initial read and the subsequent attempted write. Mind uploading the updated test without the mock, even if it isn't passing yet?

Reply:
Yes, sure!
Uploaded.

Reply:
@tritone for Java we take the existing IAM policy and reset it. Probably could be improved: https://github.com/googleapis/java-bigtable/blob/master/google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/admin/v2/it/BigtableTableAdminClientIT.java#L194-L217. I haven't seen any quota related issues so far.

Reply:
@tbpg @tritone any additional thoughts here?